src/driver/firebird/FirebirdQueryRunner.ts

```diff
@@ -0,0 +1,1651 @@
+                            this.driver.connection.logger.logQuerySlow(queryExecutionTime, query, parameters, this);
+
+                        if (err) {
+                            this.driver.connection.logger.logQueryError(err, query, parameters, this);
+                            return fail(new QueryFailedError(query, parameters, err));
+                        }
+                        ok(result);
+                    });
+                } else {
+                    this.driver.firebirdDatabase.execute(query, parameters || [], (err, result) => {
+                        const maxQueryExecutionTime = this.driver.connection.options.maxQueryExecutionTime;
+                        const queryEndTime = +new Date();
+                        const queryExecutionTime = queryEndTime - queryStartTime;
+                        if (maxQueryExecutionTime && queryExecutionTime > maxQueryExecutionTime)
+                            this.driver.connection.logger.logQuerySlow(queryExecutionTime, query, parameters, this);
+
+                        if (err) {
+                            this.driver.connection.logger.logQueryError(err, query, parameters, this);
+                            return fail(new QueryFailedError(query, parameters, err));
+                        }
+                        ok(result);
+                    });
+                }
+            } catch (err) {
+                fail(err);
+            }
+        });
+    }
+
+    /**
+     * Returns raw data stream.
+     */
+    stream(query: string, parameters?: any[], onEnd?: Function, onError?: Function): Promise<ReadStream> {
+        throw new FirebirdError("Stream query not supported");
+    }
+
+    /**
+     * Returns all available database names including system databases.
+     */
+    async getDatabases(): Promise<string[]> {
+        return Promise.resolve([]);
+    }
+
+    /**
+     * Returns all available schema names including system schemas.
+     * If database parameter specified, returns schemas of that database.
+     */
+    async getSchemas(database?: string): Promise<string[]> {
+        throw new Error(`Firebird driver does not support table schemas`);
+    }
+
+    /**
+     * Checks if database with the given name exist.
+     */
+    async hasDatabase(database: string): Promise<boolean> {
+        throw new Error(`Firebird driver does not support multiple databases connection`);
+    }
+
+    /**
+     * Checks if schema with the given name exist.
+     */
+    async hasSchema(schema: string): Promise<boolean> {
+        throw new Error(`Firebird driver does not support table schemas`);
+    }
+
+    /**
+     * Checks if table with the given name exist in the database.
+     */
+    async hasTable(tableOrName: Table|string): Promise<boolean> {
+        const parsedTableName = this.parseTableName(tableOrName);
+        const sql = `SELECT * FROM \`RDB$RELATIONS\` WHERE \`RDB$FLAGS\`= 1 AND \`RDB$RELATION_NAME\` = '${parsedTableName.tableName}'`;
```

Comment:
Here (And as well later in the code) we have the problem that not all FB versions support backticks.

```suggestion
        const sql = `SELECT * FROM "RDB$RELATIONS" WHERE "RDB$FLAGS"= 1 AND "RDB$RELATION_NAME" = '${parsedTableName.tableName}'`;
```